src/factory.rs

```diff
@@ -9,19 +9,27 @@ pub fn factory<C: RuntimeName, I: ComInterface>() -> Result<I> {
 
     unsafe {
         // First attempt to get the activation factory via the OS.
-        let mut code = RoGetActivationFactory(name.get_abi(), &I::iid(), &mut factory);
+        let code = RoGetActivationFactory(name.get_abi(), &I::iid(), &mut factory);
+
+        // Treat any delay-load errors like standard errors, so that the heuristics
+        // below can still load registration-free libraries on Windows versions below 10.
+        let mut code = code.unwrap_or_else(|code| code);
```

Comment:
nit: it's a bit strange that the functions return `Result<ErrorCode, ErrorCode>` is there any reason to not just return ErrorCode?

Reply:
This is as a result of the delay loading macro wrapping the result of the function. Ideally, it would fold the resulting error code if the result of the function happens to be an `ErrorCode` to begin with.

Reply:
Not all of the functions return ErrorCode, so the macro can't assume it.
This could use something like:

```rust
type DemandLoadResult<T> = Result<T, DemandLoadError>;
struct DemandLoadError(ErrorCode);
```

which I'd definitely do for a crate, but doesn't seem worth it for a couple of internal uses?

Reply:
Agreed. Perhaps a future enhancement.